src/main/kotlin/com/openlattice/data/storage/PostgresEntityDataQueryService.kt

```diff
@@ -874,6 +883,35 @@ class PostgresEntityDataQueryService(
 
         return WriteEvent(tombstoneVersion, numUpdated)
     }
+
+    fun getExpiringEntitiesFromEntitySet(entitySetId: UUID, sqlParams: Triple<String, Any, Int>, deleteType: DeleteType) : BasePostgresIterable<UUID> {
+        val partitionsInfo: PartitionsInfo = partitionManager.getEntitySetPartitionsInfo(entitySetId)
+        val partitions = partitionsInfo.partitions.joinToString(prefix = "('", postfix = "')", separator = "', '") { it.toString() }
+        val partitionVersion = partitionsInfo.partitionsVersion
+        return BasePostgresIterable(
+                PreparedStatementHolderSupplier(
+                        hds,
+                        getExpiringEntitiesQuery(entitySetId, sqlParams.first, deleteType, partitions),
+                        FETCH_SIZE,
+                        false
+                ) {stmt -> stmt.setInt(1, partitionVersion)
+                    stmt.setObject(2, sqlParams.second, sqlParams.third)}
+        ) { rs -> ResultSetAdapters.id(rs)}
+    }
+
+    private fun getExpiringEntitiesQuery(entitySetId: UUID, expirationBaseColumn: String, deleteType: DeleteType, partitions: String) : String {
+        var ignoredClearedEntitiesClause = ""
+        if (deleteType == DeleteType.Soft) {
+            ignoredClearedEntitiesClause = "AND ${VERSION.name} >= 0 "
+        }
+        return "SELECT ${ID.name} FROM ${PostgresTable.DATA.name} " +
+                "WHERE ${ENTITY_SET_ID.name} = '$entitySetId' " +
+                "AND ${PARTITION.name} IN $partitions " +
+                "AND ${PARTITIONS_VERSION.name} = ? " +
```

Comment:
just wondering why some of these are parameterized and some values are passed in and written to the query string?

Reply:
The last variable is the only one that is of type Any and so I have to pass with a sql type. I can make them all parameters instead if that's preferable

Reply:
Use of parameters is preferred if it's a prepared statement. SQL injection isn't a risk here, but you lose a big advantage of prepared statements due to high cardinality of possible values for entitySetId and partitions (that is each combination of passed in values generates a different prepared statement and execution plan).

Reply:
gotcha, will fix